source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +124,153 @@ In this step, we generate a local master key to expedite setup of our developmen
           }
 
 
-B. Define a JSON Schema
+B. Create a Data Encryption Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we use the MongoDB driver to create and store a data
+encryption key, used by CSFLE to automatically encrypt and decrypt data on
+document fields. The following configuration values are required by the client
+to generate a new data encryption key:
+
+* The local master key or AWS KMS master key access settings.
+* A MongoDB connection string that authenticates on a running server.
+* The key vault database and collection name.
+
+Upon creating a data encryption key, the driver returns the ``_id`` field
+(as a `UUID <https://en.wikipedia.org/wiki/Universally_unique_identifier>`_)
+of a document that stores information about the key including:
+
+* The data encryption key, encrypted by the master key.
+* The key management service provider data.
+
+Follow the steps below to generate a single data encryption key from the
+local master key.
+
+.. note::
+
+  If using AWS KMS to provide the master key, refer to the "`F. Convert to
+  a Remote Master Key`_" section for the appropriate settings.
+
+First, read the file containing the local master key generated in the "`A.
+Create a Master Key`_" section:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-master-key-reader
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String path = "master-key.txt";
+
+          byte[] fileBytes = new byte[96];
+
+          try (FileInputStream fis = new FileInputStream(path)) {
+            fileBytes = fis.readAllBytes();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+
+          final byte[] localMasterKey = Arrays.copyOf(fileBytes, 96);
+
+
+Next, specify the KMS provider settings. Since we are using a local master
+key, the provider name is ``local``:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-kms-settings-local
```

Comment:
All of our custom tab id values should be consistent, otherwise they won't stay in sync. (Eventually we should probably switch to pre-defined tabsets, but this is fine for now if we fix the id)

i.e. all the Java tabs should have the same ID, etc.

Reply:
Oops. Up to this point, I thought they should be unique. Will need to fix across multiple PRs. I'll create a JIRA ticket to do cleanup on the guide once all the sections have been merged and add this as an item to address. For now, I will use `use-cases-fle-guide-java` (use-cases section, fle guide document, language)

Reply:
Will fix.